panic detected when use moar for git diff

I use `git diff -w` with moar config in `~/.zshrc`

```zsh
# https://github.com/walles/moar
if command -v moar &>/dev/null; then
    export MOAR="-no-linenumbers -no-statusbar -quit-if-one-screen -wrap"
    export PAGER=moar
fi
```


```txt
Version: v1.24.1
LANG   : en_US.UTF-8
TERM   : xterm-kitty

GOOS    : darwin
GOARCH  : arm64
Compiler: gc
NumCPU  : 8

panic: write /dev/stdout: file already closed [recovered]
	panic: write /dev/stdout: file already closed

goroutine 1 [running]:
main.main.func1()
	github.com/walles/moar/moar.go:793 +0x150
panic({0x104bdb9e0?, 0x140003f2480?})
	runtime/panic.go:770 +0x124
github.com/walles/moar/twin.(*UnixScreen).write(...)
	github.com/walles/moar/twin/screen.go:198
github.com/walles/moar/twin.(*UnixScreen).showNLines(0x140001b0000, 0x14, 0xc8?)
	github.com/walles/moar/twin/screen.go:720 +0x2fc
github.com/walles/moar/twin.(*UnixScreen).ShowNLines(0x140001b0000?, 0x104c0ae50?)
	github.com/walles/moar/twin/screen.go:687 +0x20
github.com/walles/moar/m.(*Pager).ReprintAfterExit(0x14000282000)
	github.com/walles/moar/m/pager.go:472 +0x64
main.startPaging.func1()
	github.com/walles/moar/moar.go:851 +0x50
main.startPaging(0x14000098110?, {0x104c0e0b8?, 0x140001b0000?}, 0x104c09d88?, 0x14000180001?)
	github.com/walles/moar/moar.go:866 +0x6c
main.main()
	github.com/walles/moar/moar.go:818 +0x234
```

Dowgrade to `v1.23.5` work fine

--- a/moar.go
+++ b/moar.go
@@ -279,6 +279,7 @@ func printProblemsHeader() {
 	fmt.Fprintln(os.Stderr, "Version:", versionString)
 	fmt.Fprintln(os.Stderr, "LANG   :", os.Getenv("LANG"))
 	fmt.Fprintln(os.Stderr, "TERM   :", os.Getenv("TERM"))
+	fmt.Fprintln(os.Stderr, "MOAR   :", os.Getenv("MOAR"))
 	fmt.Fprintln(os.Stderr)
 	fmt.Fprintln(os.Stderr, "GOOS    :", runtime.GOOS)
 	fmt.Fprintln(os.Stderr, "GOARCH  :", runtime.GOARCH)

--- a/test.sh
+++ b/test.sh
@@ -77,6 +77,13 @@ echo Testing not crashing with different argument orders...
 ./moar --trace +123 moar.go >/dev/null
 ./moar --trace moar.go +123 >/dev/null
 
+# We can only do this test if we have a terminal. This means it will be run
+# locally but not in CI. Not great, but better than nothing.
+if [[ -t 1 ]]; then
+  echo Test auto quitting on single screen...
+  echo "  (success)" | ./moar --quit-if-one-screen
+fi
+
 echo Test decompressing while piping
 # Related to https://github.com/walles/moar/issues/177
 ./moar sample-files/compressed.txt.gz | grep compressed >/dev/null

--- a/twin/screen-setup.go
+++ b/twin/screen-setup.go
@@ -45,18 +45,27 @@ func (screen *UnixScreen) setupSigwinchNotification() {
 }
 
 func (screen *UnixScreen) setupTtyInTtyOut() error {
+	// Dup stdout so we can close stdin in Close() without closing stdout.
+	// Before this dupping, we crashed on using --quit-if-one-screen.
+	//
+	// Ref:https://github.com/walles/moar/issues/214
+	stdoutDupFd, err := syscall.Dup(int(os.Stdout.Fd()))
+	if err != nil {
+		return err
+	}
+	stdoutDup := os.NewFile(uintptr(stdoutDupFd), "moar-stdout-dup")
+
 	// os.Stdout is a stream that goes to our terminal window.
 	//
 	// So if we read from there, we'll get input from the terminal window.
 	//
-	// Reading from os.Stdin will fail if we're getting data piped into
-	// ourselves from some other command.
+	// If we just read from os.Stdin that would fail if we're getting data piped
+	// into ourselves from some other command.
 	//
 	// Tested on macOS and Linux, works like a charm!
-	screen.ttyIn = os.Stdout // <- YES, WE SHOULD ASSIGN STDOUT TO TTYIN
+	screen.ttyIn = stdoutDup // <- YES, WE SHOULD ASSIGN STDOUT TO TTYIN
 
 	// Set input stream to raw mode
-	var err error
 	screen.oldTerminalState, err = term.MakeRaw(int(screen.ttyIn.Fd()))
 	if err != nil {
 		return err

